languages/python/oso/polar/exceptions.py

```diff
@@ -3,71 +3,163 @@
 # @TODO: Should we just generate these from the rust code?
 
 
-class PolarException(Exception):
-    """Base class for all exceptions from within polar."""
+class OsoError(Exception):
+    def __init__(self, message=None, details=None):
+        self.message = message
+        self.details = details
+        self.stack_trace = details.get("stack_trace") if details else None
+        super().__init__(self.message)
 
-    def __init__(self, message, error=None):
-        super(PolarException, self).__init__(message)
-        self._inner = error
 
+class FFIErrorNotFound(OsoError):
```

Comment:
Could we add documentation strings to this module & exception types indicating when they would occur?